lib/events/index.js

```diff
@@ -2042,6 +2101,11 @@ export default function (self) {
 
     return rows.length;
   };
+  self.pasteDataReplicate = function (data, dataType, remaining) {
```

Comment:
This might not be entirely finished yet (so ignore this comment if so), and I may be misunderstanding the intent here, but this looks like a function intended to provide the values for the cells that are being 'filled down' (in Excel parlance) — if that's so I don't think `pasteDataReplicate` would be an entirely proper name, because we could, as users, fill down just by dragging the handle, not even pasting the data.

That said, I believe we would need to provide the user an ability to override the default fill down behavior (if we even want to specify a default fill down behavior — what do you think?) using a configurable attribute in the form of a function, or maybe an iterator or (perhaps better) a [generator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators#generator_functions). 

See the `Parameters & Attributes` in the [documentation](https://canvas-datagrid.js.org/) for examples of attributes.

To illustrate, I'm thinking something along the lines of:

```js
const grid = canvasDatagrid({
  fillDownFunction: function getFillDownData(data, dataType, ...) {
     return [1, 2, 3];
  }

  // or:

  fillDownFunction: function* makeFillDownIterator(data, dataType, ...) {
    if (isNumericData(data) === false) {
      return []; // we cannot produce the next value if the it's not a numeric sequence
    }

    let lastValue = data[data.length - 1];
    let newValue = lastValue + 1;

    while (true) {
      yield newValue++;
    }
  }
})
```

We could determine whether `fillDownFunction` is a normal function or a generator function by checking its constructor, and acting accordingly.

Note: I'm not entirely sure what arguments would be passed to such a function, hence the `...` notation — but I'd imagine the filling would have to behave differently not just based on the data is being pasted or filled down, but _where_ in the sheet it is being pasted or filled down, so something like `columnIndex/rowIndex` (and `viewRowIndex/viewColumnIndex`) would probably be useful to pass along. Selection size/bounds, as well, maybe?

If the number of positional arguments to such a function becomes too large (let's say more than 3 or 4), I would propose we use an object instead (whatever its contents):

```ts
interface FillDownFunctionProps {
  // ...
}

fillDownFunction: function *makeFillDownIterator(props: FillDownFunctionProps) {
  // ...
}
```

This is all a bit loosely sketched out / thinking out loud — what do you think?

Reply:
Hi,

I passed an object instead, and I think it receives all the necessary arguments to easily transform cell data before placing it. What do you think?